### ISSUE TITLE
Semantics of Permissions, Prohibitions, and Obligations

### ISSUE THREAD

**Issue by AndreaCimminoArriaga** [w3c/odrl#74]
The [semantics of Permissions, Prohibitions, and Obligations](https://w3c.github.io/odrl/formal-semantics/#:~:text=3.-,Semantics%20of%20Permissions%2C%20Prohibitions%2C%20and%20Obligations,-The%20deontic%20modalities) from a descriptive point of view are quite clear. However, IMHO, from the evaluation point of view they do not have clear differences. At the end, a policy encodes a set of rules, each of which codifies a set of restrictions and an action to be performed. 

It is true that Obligations may require a proof from a third-party actor, however, I wonder if such proof could not be encoded as an action itself hiding how this proof is obtained and checked in the policy and leaving such behaviour to the implementations.  Maybe the difference, from the implementation point of view, is that permissions and prohibitions are usually perceived as synchronous evaluations of the policies; a requester wants to do something with a resource, a set of suitable policies are found, they are evaluated and their actions performed. Instead, Obligations are more asynchronous alike, something happens in the world, the policy then needs some data-related to such event, and depending on that an action happens

**Comment by fornaran** [comment]
I agree with the content of this comment and we have to consider it as soon as we will discuss in details the semantics of prohibitions, obligations, and the semantics of a policy with permissions, prohibitions, and obligations.

When the focus is on the semantics of **obligations**,  I agree that it is correct to say that a State S complies with an obligation if there **exists** in the State an action a (called fulfilling action) that satisfies a set of conditions (e.g. the obligation is active, the action a has the required type, i.e. the type of the action regulated by the obligation, refinements of the regulated action are satisfied by the existing action a, target and assignee are the correct one, as specified in the obligation with their refinements).

Differently when we speak about **permissions** and **prohibitions** we need to check that **all** the actions in the state S comply with at least one permission and either the conflict strategy of the policy  is "perm" or all the actions in state S comply with all the prohibitions (all the actions in state S are permitted and are not prohibited). We need also to use the **behaviour** attribute for those actions that are neither permitted nor prohibited.
